packages/react-components/src/components/knowledge-graph/KnowledgeGraphPanel.tsx

```diff
@@ -1,24 +1,117 @@
-import React, { useEffect, useCallback, useMemo, useState } from 'react';
+import React, { HTMLAttributes, useEffect, useCallback, useMemo, useState, useRef } from 'react';
 import { IntlProvider, FormattedMessage } from 'react-intl';
-import { ElementDefinition } from 'cytoscape';
+import type { Core, ElementDefinition, EventObjectNode, EventObjectEdge } from 'cytoscape';
 import { Button, Container, Header, Input, SpaceBetween } from '@cloudscape-design/components';
 import { TwinMakerKGQueryDataModule } from '@iot-app-kit/source-iottwinmaker';
-import { Graph } from './graph';
+import GraphView from './graph/graph-view';
+import Toolbar from './graph/graph-toolbar';
+import './graph/styles.scss';
+import { STYLE_PREFIX } from './graph/constants';
+import useStylesheet from './graph/cytoscape-cloudscape-theme';
 import StateManager, { useKnowledgeGraphState } from './StateManager';
 import { createKnowledgeGraphQueryClient } from './KnowledgeGraphQueries';
 import { ResponseParser } from './responseParser';
+import { NodeData, EdgeData } from './graph/types';
+import { IQueryData } from './interfaces';
 import { getElementsDefinition } from './utils';
 
-interface KnowledgeGraphInterface {
+interface KnowledgeGraphInterface extends HTMLAttributes<HTMLDivElement> {
   kgDataSource: TwinMakerKGQueryDataModule;
+  onEntitySelected?: (e: NodeData) => void;
+  onRelationshipSelected?: (e: EdgeData) => void;
+  onEntityUnSelected?: (e: NodeData) => void;
+  onRelationshipUnSelected?: (e: EdgeData) => void;
+  queryData?: IQueryData;
 }
 const MAX_NUMBER_HOPS = 10;
+export const ZOOM_INTERVAL = 1;
 
-const KnowledgeGraphContainer: React.FC<KnowledgeGraphInterface> = ({ kgDataSource }) => {
-  const { selectedGraphNodeEntityId, setQueryResult, queryResult, clearGraphResults } = useKnowledgeGraphState();
+const KnowledgeGraphContainer: React.FC<KnowledgeGraphInterface> = ({
+  kgDataSource,
+  className,
+  onEntitySelected,
+  onRelationshipSelected,
+  onEntityUnSelected,
+  onRelationshipUnSelected,
+  queryData,
+  ...props
+}) => {
+  const cy = useRef<Core>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const stylesheet = useStylesheet(containerRef);
+  const { selectedGraphNodeEntityId, setSelectedGraphNodeEntityId, setQueryResult, queryResult, clearGraphResults } =
+    useKnowledgeGraphState();
   const [searchTerm, setSearchTerm] = useState('');
   const [elements, setElements] = useState<ElementDefinition[]>([]);
 
+  const fit = useCallback(() => {
+    cy.current?.fit();
+  }, []);
+
+  const center = useCallback(() => {
+    cy.current?.center();
+  }, []);
+
+  const zoomIn = useCallback(() => {
+    cy.current?.zoom(cy.current.zoom() + ZOOM_INTERVAL);
+  }, []);
+
+  const zoomOut = useCallback(() => {
+    cy.current?.zoom(cy.current.zoom() - ZOOM_INTERVAL);
+  }, []);
+
+  useEffect(() => {
+    cy.current?.resize();
+    cy.current?.center();
+  }, []);
```

Comment:
after testing I notices these lines have no effect at all when using cytoscape-reactjs library, very strange. I deleted them for now. I set fit to true for the demo and I left true however provide undesired fitting with every render i feel this should be left for the user for example if you zoom the graph the select a node it will re fit immediately. It turns out node repulsion has to be set explicitly also this strange behavior only happens in cytoscape-reactjs library. In pur cytoscape these values are default and get applied and doesn't need to be explicitly set. Anyway the layout render correctly now and I removed the above useEffect.